qt-build/src/lib.rs

```diff
@@ -0,0 +1,254 @@
+// SPDX-FileCopyrightText: 2022 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+// SPDX-FileContributor: Be Wilson <be.wilson@kdab.com>
+//
+// SPDX-License-Identifier: MIT OR Apache-2.0
+
+//! This crate provides information about the Qt installation and can invoke Qt's
+//! [moc](https://doc.qt.io/qt-6/moc.html) code generator. This crate does not build
+//! any C++ code on its own. It is intended to be used in [build.rs scripts](https://doc.rust-lang.org/cargo/reference/build-scripts.html)
+//! together with
+//! [cc](https://docs.rs/cc/latest/cc/),
+//! [cxx_build](https://docs.rs/cxx-build/latest/cxx_build/), or
+//! [cpp_build](https://docs.rs/cpp_build/latest/cpp_build/).
+
+use std::{
+    env,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+pub use versions::SemVer;
+
+/// Helper for build.rs scripts using Qt
+/// ```
+/// let qt_modules = vec!["Core", "Gui"]
+///     .iter()
+///     .map(|m| String::from(*m))
+///     .collect();
+/// let qtbuild = qt_build::QtBuild::new(qt_modules).expect("Could not find Qt installation");
+/// ```
+pub struct QtBuild {
+    pub version: SemVer,
```

Comment:
this field doesn't need to be `pub` ?

Reply:
Downstream users might want to have some code conditional on the Qt version.

Reply:
returning a const reference is a better idea